lib/autobuild/import/git.rb

```diff
@@ -1260,13 +1287,24 @@ def checkout(package, _options = Hash.new)
             FileUtils.mkdir_p(base_dir) unless File.directory?(base_dir)
 
             clone_options = Array.new
-            clone_options << '--recurse-submodules' if with_submodules?
+            if with_submodules?
+                clone_options << '--recurse-submodules'
+                clone_options << '--shallow-submodules' if shallow?
+            end
+
+            clone_options << '--depth' << '1' if shallow?
+
             if single_branch?
-                if remote_branch.start_with?("refs/")
-                    raise ArgumentError, "you cannot provide a full ref for"\
-                        " the remote branch while cloning a single branch"
+                if tag
+                    clone_options << "--branch=#{tag}"
```

Comment:
> it works only if the tag is not fully qualified (i.e. refs/tags/v2.9.0 does not work, only v2.9.0)

Didn't realize that. Thanks.

> it leaves the repo with a detached HEAD

The problem with `tag` and `single_branch` is that if you don't clone the branch that contains the tag, you don't get the tag ref. Passing the tag to `--branch` works around that. Now, that does introduce the side effect of leaving the repo with a detached HEAD and that's why I call `#update_remotes_configuration` with `only_local: false` later on, which makes the importer set up a branch. Since we are cloning, `only_local: false` sounds reasonable to me. Could you please give this a try in one of your workspaces and let me know what do you think?

Reply:
Hi @g-arjones. Thanks for the explanation. Sound reasonable to me. Please put this explanation either in the commit message or in the code itself, it would help understand these (delicate) things later.